src/sentry/notifications/helpers.py

```diff
@@ -461,14 +501,65 @@ def get_most_specific_notification_setting_value(
     """
     return (
         get_highest_notification_setting_value(
-            notification_settings_by_scope.get(get_scope_type(type), {}).get(parent_id, {})
+            get_value_for_parent(notification_settings_by_scope, parent_id, type)
         )
         or get_highest_notification_setting_value(
-            (
-                notification_settings_by_scope.get(NotificationScopeType.USER)
-                or notification_settings_by_scope.get(NotificationScopeType.TEAM)
-                or {}
-            ).get(recipient.id, {})
+            get_value_for_actor(notification_settings_by_scope, recipient)
         )
         or _get_notification_setting_default(ExternalProviders.EMAIL, type)
     )
+
+
+def has_any_non_never_settings(
+    notification_settings_by_scope: Mapping[
+        NotificationScopeType,
+        Mapping[int, Mapping[ExternalProviders, NotificationSettingOptionValues]],
+    ],
+    recipient: Team | User,
+    parent_id: int,
+    type: NotificationSettingTypes,
+) -> bool:
+    """
+    Given notification settings by scope, an organization or project, and a
+    notification type, will a recipient receive any notifications?
+    """
+    return any(
+        value != NotificationSettingOptionValues.NEVER
+        for value in get_values_by_provider(
+            notification_settings_by_scope, recipient, parent_id, type
+        ).values()
+    )
+
+
+def merge_notification_settings_up(
+    *settings_mappings: Mapping[ExternalProviders, NotificationSettingOptionValues],
+) -> Mapping[ExternalProviders, NotificationSettingOptionValues]:
+    """
+    Given a list of notification settings by provider ordered by increasing
+    specificity, get the most specific value by provider.
+    """
+    value_by_provider: MutableMapping[ExternalProviders, NotificationSettingOptionValues] = {}
```

Comment:
I'm having a hard time understanding what's happening here, can you give an example of an input and the changed output? Where is the ordering by increasing specificity happening?

Reply:
```py
a = {'x': 1, 'z': 1}
b = {'x': 2, 'y': 2}

 # {'x': 2, 'y': 2, 'z': 1}
merge_notification_settings_up(a, b)

 # {'x': 1, 'y': 2, 'z': 1}
merge_notification_settings_up(b, a)
```

Reply:
Okay I'm still a little confused, `a` in this case is always whatever default values we've hardcoded, and then `b` is all of the user's settings, is that correct? So as a concrete example I ran [test_get_setting_mapping_from_mapping_project](https://github.com/getsentry/sentry/blob/73294e450a8cd61340b50ec340123c7901ec3286/tests/sentry/notifications/utils/test_get_setting_mapping_from_mapping.py#L118) which I believe has user settings that are the same as the problem this PR is solving and then printed this out:
```
default_settings = {EMAIL: ALWAYS, SLACK: NEVER}
user_settings = {EMAIL: NEVER: 10, SLACK: ALWAYS}, {}, {EMAIL: NEVER}
result = merge_notification_settings_up(default_settings, user_settings)
result: {EMAIL: NEVER, SLACK: ALWAYS}
```

I guess this isn't the most helpful example because there is no change though. So is the default value being passed so that it can be used if there happen to be no settings for a given provider? I also still don't understand where the ordering by increasing specificity is happening.

To sum up: beforehand the project level `{EMAIL: NEVER}` was taking priority over the user level `{SLACK: ALWAYS}`, causing the group subscription (the bell icon) to be disabled, and now we are instead (I need to save this comment and re-read the changes) checking the user level settings?